src/main/java/spireMapOverhaul/zones/heavenlyClouds/HeavenlyCloudsZone.java

```diff
@@ -76,6 +101,106 @@ public void atBattleStart() {
         }
     }
 
+    private class Cloud {
+        private Texture texture;
+        private float x, y;
+        private float speed;
+        private float scale;
+        private Color color;
+
+        public Cloud(Texture texture, float x, float y, float speed, float scale, Color color) {
+            this.texture = texture;
+            this.x = x;
+            this.y = y;
+            this.speed = speed;
+            this.scale = scale;
+            this.color = color;
+        }
+    }
+
+    @Override
+    public void postRenderCombatBackground(SpriteBatch sb) {
+        if (!initialCloudSpawned) {
+            spawnInitialClouds();
+            initialCloudSpawned = true;
+        }
+
+        cloudSpawnTimer += Gdx.graphics.getDeltaTime();
+        if (cloudSpawnTimer >= cloudSpawnInterval) {
+            cloudSpawnTimer = 0;
+            spawnCloud();
+        }
+
+        Iterator<Cloud> iter = clouds.iterator();
+        while (iter.hasNext()) {
+            Cloud cloud = iter.next();
+            cloud.x += cloud.speed * Gdx.graphics.getDeltaTime();
+            if (cloud.x > Settings.WIDTH) {
+                iter.remove();
+            } else {
+                sb.setColor(cloud.color);
+                sb.draw(cloud.texture, cloud.x, cloud.y, cloud.texture.getWidth() * cloud.scale, cloud.texture.getHeight() * cloud.scale);
+            }
+        }
+
+        sb.setColor(Color.WHITE);
+    }
+
+
+    private void spawnCloud() {
+        Texture cloudTexture;
+        boolean isSpecialCloud = MathUtils.random() < 0.01f; // 1% chance to spawn a special cloud
+        if (isSpecialCloud) {
+            cloudTexture = specialClouds[MathUtils.random(specialClouds.length - 1)];
+        } else {
+            Texture[] cloudTextures = {Cloud1, Cloud2, Cloud3, Cloud4, Cloud5};
+            cloudTexture = cloudTextures[MathUtils.random(cloudTextures.length - 1)];
+        }
+        float x = -cloudTexture.getWidth() - 5;
```

Comment:
There's a lot of overlap between this and initial cloud spawning, maybe extract the duplicate code into a method to reduce repeats